src/prepare_dlc_dev_environment.py

```diff
@@ -259,6 +265,189 @@ def commit_and_push_changes(changes, remote_push=None, restore=False):
+    short_version values updated accordingly.
+    """
+    buildspec_pattern = (
+        r"^(\w+)/(training|inference)/buildspec(?:-(\w+))?-(\d+)-(\d+)(?:-(.+))?\.yml$"
+    )
+    for currency_path in currency_paths:
+        if not validate_currency_path(currency_path):
+            continue
+        (
+            framework,
+            job_type,
+            optional_suffix,
+            major_version,
+            minor_version,
+            extra_suffix,
+        ) = extract_path_components(currency_path, buildspec_pattern)
+
+        latest_version_path = find_latest_version_path(
+            framework, job_type, optional_suffix, major_version, extra_suffix
+        )
+        if latest_version_path:
+            updated_content = generate_new_file_content(
+                latest_version_path, major_version, minor_version, optional_suffix, extra_suffix
+            )
+            create_new_file_with_updated_version(
+                currency_path, updated_content, optional_suffix, extra_suffix, latest_version_path
+            )
+        else:
+            LOGGER.warning(f"No previous version found for {currency_path}")
+
+
+def find_latest_version_path(framework, job_type, optional_suffix, major_version, extra_suffix):
+    """
+    Finds the path to the latest existing version of the buildspec file based on the provided components.
+    """
+    path_prefix = os.path.join(get_cloned_folder_path(), framework, job_type)
+    version_pattern = r"buildspec(?:-{})?-{}-(\d+)(?:-{})?\.yml".format(
+        optional_suffix or r"\w*", major_version, extra_suffix or r"\w*"
+    )
+    latest_version = Version("0.0.0")
+    latest_path = None
+
+    for file_name in os.listdir(path_prefix):
+        match = re.match(version_pattern, file_name)
+        if match:
+            version_str = match.group(1)
+            version = Version(f"{major_version}.{version_str}.0")
+            if version > latest_version:
+                latest_version = version
+                latest_path = os.path.join(path_prefix, file_name)
+
+    return latest_path
+
+
+def validate_currency_path(currency_path):
+    """
+    Validates the currency path format using the provided regular expression pattern, and returns
+    true if the currency path format is valid, False otherwise. If an extra framework is detected,
+    it prints an error message and returns False.
+    """
+    buildspec_pattern = (
+        r"^(?:(\w+)/)?(\w+)/(training|inference)/buildspec(?:-(\w+))?-(\d+)-(\d+)(?:-(.+))?\.yml$"
+    )
+    match = re.match(buildspec_pattern, currency_path)
+    if match:
+        (
+            extra_framework,
+            framework,
+            job_type,
+            optional_suffix,
+            major_version,
+            minor_version,
+            extra_suffix,
+        ) = match.groups()
+        if extra_framework:
+            LOGGER.error(
+                f"ERROR: {extra_framework} is not currently supported. Please provide valid frameworks."
```

Comment:
Can you explicitly mention that this framework is not supported for the currency feature? Otherwise, users might think they can't use it for normal buildspec operations, which they can